### ISSUE TITLE
Fix Eval performece hit

### ISSUE THREAD

**Issue by ron-popov** [geofs-plugins/plugin-manager-V2#13]
How am i supposed to take a string representing an extension file (like the one in ```ext.user.js```) and turn it into an object i can read data from and call it's methods ?
@yotam180

**Comment by yotam180** [comment]
You can run `eval()`...?

**Comment by ron-popov** [comment]
Does it return the extension object ?

**Comment by yotam180** [comment]
Yeah. Because it's the last statement of code (actually the only one) `eval()` should return the object returned by the invoked anonymous function. If it doesn't, reopen this issue and tell me. I'll find out why.

**Comment by ron-popov** [comment]
Neat, i think it works just fine i just didn't use it correctly :P

**Comment by yotam180** [comment]
You know what, I think there's a better way. You should provide a method `registerPlugin()` or something, that instead of returning `ext`, the anonymous function would call the method `registerPlugin(ext);`. Then instead of evaluating the script, you add it as a `<script>` tag into the HTML.
It would improve the performance as `eval` is slow.

**Comment by ron-popov** [comment]
Ok, i will first get a working version then will do that.
